packages/material-ui-lab/src/Autocomplete/Autocomplete.test.js

```diff
@@ -37,10 +37,17 @@ describe('<Autocomplete />', () => {
         <Autocomplete {...defaultProps} renderInput={(params) => <TextField {...params} />} />,
       );
       const input = getByRole('textbox');
-      input.focus();
-      fireEvent.change(document.activeElement, { target: { value: 'a' } });
+
+      act(() => {
+        input.focus();
+        fireEvent.change(document.activeElement, { target: { value: 'a' } });
+      });
```

Comment:
I thought testing library already included act when firing events

```suggestion
      act(() => {
        input.focus();
      });
      fireEvent.change(document.activeElement, { target: { value: 'a' } });
```

Reply:
Yeah it does. That doesn't necessarily mean that we can't put these in a single `act()` because that's a difference in behavior:

```js
act(one);
// two runs with updates flushed from one
act(two)
```

vs
```js
act(() => {
  one();
  // two runs without all updates flushed from one (I think)
  two();
})
```

I'm not 100% certain these are different. Though I would expect that idle or offscreen updates scheduled in `one` are not necessarily flushed before `two` runs.

I'll try to wrap actions for an assertion in a single act() i.e. between each act() should be an assertion. 

In the end wrapping it in a single act() is closer to the original behavior which is more important here.

Reply:
Interesting distinction, I will keep it in mind. So, in the end, we should consider if two actions will happen during the same screen, in between 16ms. If it's two consequent events, like mouseup/click, it should be wrapped in the same act, if it's two user actions, like key down + key up, it should be wrapped in two act.

Reply:
I'm not sure to be honest. It seems to me that as few `acts` as possible should be preferred because the "time" between two acts() can be very long